src/multiinstances/internal/ipc/ipcloop.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "ipcloop.h"
+
+using namespace mu::ipc;
+
+IpcLoop::IpcLoop()
+{
+    m_timer.setSingleShot(true);
+    QObject::connect(&m_timer, &QTimer::timeout, [this]() {
+        exit(Code::Timeout);
+    });
+}
+
+Code IpcLoop::exec(int timeout)
+{
```

Comment:
might make sense to protect it with m_timer.isActive()

Reply:
added